pkg/image/importer/credentials.go

```diff
@@ -0,0 +1,112 @@
+package importer
+
+import (
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"github.com/docker/distribution/registry/client/auth"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+var (
+	NoCredentials auth.CredentialStore = &noopCredentialStore{}
+
+	emptyKeyring = &credentialprovider.BasicDockerKeyring{}
+)
+
+type noopCredentialStore struct{}
+
+func (s *noopCredentialStore) Basic(url *url.URL) (string, string) {
```

Comment:
Line 18 is a type assertion

On Wed, Jan 20, 2016 at 3:18 PM, David Eads <notifications@github.com>
wrote:

> In pkg/image/importer/credentials.go
> <https://github.com/openshift/origin/pull/6288#discussion_r50312558>:
>
> > +	"github.com/docker/distribution/registry/client/auth"
> > +
> > +	kapi "k8s.io/kubernetes/pkg/api"
> > +	"k8s.io/kubernetes/pkg/credentialprovider"
> > +	"k8s.io/kubernetes/pkg/util"
> > +)
> > +
> > +var (
> > +	NoCredentials auth.CredentialStore = &noopCredentialStore{}
> > +
> > +	emptyKeyring = &credentialprovider.BasicDockerKeyring{}
> > +)
> > +
> > +type noopCredentialStore struct{}
> > +
> > +func (s *noopCredentialStore) Basic(url *url.URL) (string, string) {
>
> I'm assuming this is for
> docker/distribution/registry/client/auth.CredentialStore, but that's not
> a commonly known one. Type assertions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6288/files#r50312558>.
>